lib/middleware.js

```diff
@@ -48,6 +49,15 @@ exports.methodNotAllowed = function (allow) {
   };
 };
 
+exports.timeout = function () {
```

Comment:
do we want to add any log statements here ?

Reply:
+1 for log but location, i have pointed in the below comment. No need to log it for each request only on timeout it can be logged.

Reply:
Agree. We should have some UT & possibly even Acceptance test. Using `sinon.FakeTimers` we can fake the elapsed time and probably catch it in UT and even some acceptance test. Also, did we test the actual bug and did CC stop firing in additional requests? In other words have we verified the actual duplicate deployment scenario.

Reply:
Ok.. will work on adding an UT/AT for this..

Reply:
For the part of testing if this solves the duplicate deployment scenario, please refer to the Test results comment below. (Point 2)